src/pyseekdb/utils/embedding_functions/bm25_sparse_embedding_function.py

```
@@ -247,66 +143,91 @@ class BM25SparseEmbeddingFunction(SparseEmbeddingFunction):
         avg_doc_length: float = DEFAULT_AVG_DOC_LENGTH,
-        token_max_length: int = DEFAULT_TOKEN_MAX_LENGTH,
+        dim: int = DEFAULT_DIM,
+        language: str = "english",
         stopwords: Iterable[str] | None = None,
     ) -> None:
+        try:
+            import bm25s  # noqa: F401
+        except ImportError as exc:
+            raise ValueError(
+                "The bm25s package is not installed. Please install it with `pip install bm25s[full]`"
+            ) from exc
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)